src/Microsoft.Extensions.OptionsModel/OptionsServiceCollectionExtensions.cs

```diff
@@ -21,6 +20,8 @@ public static IServiceCollection AddOptions(this IServiceCollection services)
             }
 
             services.TryAdd(ServiceDescriptor.Singleton(typeof(IOptions<>), typeof(OptionsManager<>)));
+            // REVIEW: what do we want the lifetime to be?  TODO: make this scoped and try to automatically do IDisposable tracking
```

Comment:
That's an interesting idea. I don't think you could completely solve leaks by making it scoped - because components might request it and each have different lifetimes.